src/main/scala/clocks/Bundles.scala

```diff
@@ -0,0 +1,16 @@
+// See LICENSE.SiFive for license details.
+package freechips.rocketchip.clocks
+
+import chisel3._
+import freechips.rocketchip.util.HeterogeneousBag
+
+class ClockBundle(val params: ClockBundleParameters) extends Bundle
+{
+  val clock = Clock()
+  val reset = Bool()
+}
+
+class ClockGroupBundle(val params: ClockGroupBundleParameters) extends Bundle
+{
+  val member = HeterogeneousBag(params.members.map(p => new ClockBundle(p)))
+}
```

Comment:
Why does this have to be a `HeterogeneousBag` and not `Seq[ClockBundle]`?

Reply:
"Have to" is a bit strong but the idea is that you are making a single port that has many clocks, but each clock bundle may have different edge parameters from the others (unlike `Vec`). At the moment it is a bit moot since nothing about the edge parameters actually effects the clock bundle wires themselves, but if we add e.g. optional clock enables in the future, this would let some members have it and some not. As opposed to `Seq`, HB is an actual hardware Record type, so you can make a single port that actually has that Data type, put it in a different container`[T<:Data]`, etc. (Whereas `Seq` is not `<: Data`).

Reply:
> "Have to" is a bit strong but the idea is that you are making a single port that has many clocks, but each clock bundle may have different edge parameters from the others (unlike `Vec`). At the moment it is a bit moot since nothing about the edge parameters actually effects the clock bundle wires themselves, but if we add e.g. optional clock enables in the future, this would let some members have it and some not. As opposed to `Seq`, HB is an actual hardware Record type, so you can make a single port that actually has that Data type, put it in a different container`[T<:Data]`, etc. (Whereas `Seq` is not `<: Data`).

now it becomes `val member: RecordMap[ClockBundle] = RecordMap(params.members.map { case (k, v) =>
    k -> new ClockBundle(v)
  })`
 _As opposed to Seq, HB is an actual hardware Record type, so you can make a single port that actually has that Data type, put it in a different container[T<:Data], etc. (Whereas Seq is not <: Data)._
I still felt confused about this, you mean I can make a single port that actually has HeterogeneousBag(Or RecordMap) type? what do you mean by  put it in a different container[T<:Data], etc.